src/Pages/7 Writing Plugins/Testing.md

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Testing Your Plugin'
+date: 2025-01-15
+image: images/blake-logo.png
+tags: []
+description: "Comprehensive testing strategies for Blake plugins including unit tests, integration tests, and development tips."
+iconIdentifier: "bi bi-plus-square-fill-nav-menu"
+pageOrder: 8
+category: "Writing Plugins"
+---
+
+:::summary
+Test Blake plugins with unit tests for plugin logic, integration tests with actual Blake content, and thorough development testing across various content types and configurations.
+:::
+
+## Testing Your Plugin
+
+### Unit Testing
+
+Test your plugin logic independently of Blake:
+
+```csharp
+[Fact]
+public async Task LastModifiedPlugin_AddsTimestampMetadata()
+{
+    // Arrange
+    var plugin = new LastModifiedPlugin();
+    var context = CreateTestBlakeContext();
+    var logger = new TestLogger();
+
+    // Act
+    await plugin.BeforeBakeAsync(context, logger);
+
+    // Assert
+    Assert.All(context.Pages, page =>
+        Assert.True(page.Metadata.ContainsKey("lastModified")));
+}
+```
+
+**Creating Test Contexts:**
+
+```csharp
+private BlakeContext CreateTestBlakeContext()
+{
+    var pages = new List<PageModel>
+    {
+        new PageModel
+        {
+            Slug = "test-page",
+            Content = "# Test Page\n\nThis is test content.",
+            SourcePath = "/test/test-page.md",
+            Metadata = new Dictionary<string, string>()
+        }
+    };
+
+    return new BlakeContext
+    {
+        Pages = pages,
+        ProjectPath = "/test",
+        GeneratedContent = new List<GeneratedPage>()
+    };
+}
+```
+
+**Testing Error Conditions:**
+
+```csharp
+[Fact]
+public async Task Plugin_HandlesEmptyContent_Gracefully()
+{
+    // Arrange
+    var plugin = new MyPlugin();
+    var context = new BlakeContext
+    {
+        Pages = new List<PageModel>
+        {
+            new PageModel { Content = "", Slug = "empty" }
+        }
+    };
+
+    // Act & Assert - Should not throw
+    await plugin.BeforeBakeAsync(context, new TestLogger());
+}
+
+[Fact]
+public async Task Plugin_HandlesNullLogger_Gracefully()
+{
+    // Arrange
+    var plugin = new MyPlugin();
+    var context = CreateTestBlakeContext();
+
+    // Act & Assert - Should not throw
+    await plugin.BeforeBakeAsync(context, null);
+}
+```
+
+### Integration Testing
+
+Test your plugin with actual Blake content:
+
+1. **Create a minimal Blake site for testing**
+2. **Install your plugin via project reference**
+3. **Run `blake bake` and verify output**
+4. **Use automated tests to verify generated content**
+
+**Example Integration Test:**
+
+```csharp
+[Fact]
+public async Task Plugin_IntegrationTest_WithRealBlakeSite()
```

Comment:
Note that Blake doesn't load plugins by DLL alone - they still have to be referenced in the target project, either as a nuget package ref or a project ref. So this means even if you copy the DLL, restore will fail if you don't have it available as one of these two sources.

So probably need a recommendation here to either add your plugin project as a project ref to a target test site, or create a local nuget package feed.